alloc-fmt/src/lib.rs

```diff
@@ -64,6 +64,7 @@ pub struct FDWriter(pub libc::c_int);
 
 impl Write for FDWriter {
     #[inline]
+    #[cfg(not(target_os = "windows"))]
```

Comment:
Instead of making two copies of the function, you should be able to just do the following inside the `unsafe` block:

```rust
#[cfg(not(target_os = "windows"))]
let written = libc::write(self.0, buf.as_ptr() as *const _, buf.len());
#[cfg(target_os = "windows")]
let written = libc::write(self.0, buf.as_ptr() as *const _, buf.len() as libc::c_uint);
```